Neos.Media/Classes/Domain/Model/ThumbnailGenerator/ImageThumbnailGenerator.php

```diff
@@ -80,4 +81,14 @@ public function refresh(Thumbnail $thumbnail)
             throw new Exception\NoThumbnailAvailableException($message, 1433109654, $exception);
         }
     }
+
+    /**
+     * @param PersistentResource $resource
+     * @param array $adjustments
+     * @return array
+     */
+    protected function processResource(PersistentResource $resource, array $adjustments)
+    {
+        return $this->imageService->processImage($resource, $adjustments);
+    }
```

Comment:
That's what I'm saying. We don't need the AOP of the refresh method anymore like previously. 
But without this change in a custom generator you need to override a lot more code. Now you only need to override `processResource` in most cases and you can do what you want.